include/acl.h

```diff
@@ -496,6 +503,11 @@ typedef struct acl_device_def_autodiscovery_t {
   std::array<acl_system_global_mem_def_t, ACL_MAX_GLOBAL_MEM> global_mem_defs;
 
   std::vector<acl_hostpipe_info_t> acl_hostpipe_info;
+
+  // Device global definition.
+  unsigned int num_device_global;
```

Comment:
Apologies for missing this the first time: Can `num_device_global` be removed, in favour of using `device_global_mem_defs.size()` which is `O(1)` in time? The idea is to make as much use as possible of the self-describing properties of C++ containers.

Reply:
Thanks! Done.

Reply:
Perfect. I think you forgot to remove the member. 
```suggestion
```

Reply:
Thanks for removing it. Sorry about that 🤦